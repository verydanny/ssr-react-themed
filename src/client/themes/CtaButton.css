@value ( RentBlue ) from 'ui/shared/themes/colors/colors.css';

@value ( BasicShadow, emailIcon, phoneIcon ) from 'ui/shared/themes/variables/variables.css';

.CtaButton {
  display: flex;
  align-items: center;
  background: RentBlue;
  box-shadow: BasicShadow;
  border-radius: 4px;
  background-size: 18px;
  background-repeat: no-repeat;
  background-position: 10px center;
  padding: 0 0 0 35px;
  font-size: 11px;
  color: #fff;
  text-decoration:none;
}

.CtaButton-email {
  background-image: emailIcon;
  border: 0px;
}


.CtaButton-phone {
  background-image: phoneIcon;
}

.CtaButton-phone-list,
.CtaButton-email-list,
.CtaButton-email-list-center {
  height: 25px;
}

.CtaButton-phone-map,
.CtaButton-email-map,
.CtaButton-email-map-center {
  width: 96px;
  height: 28px;
  box-sizing: content-box;
}

.CtaButton-phone-list {
  margin-bottom: 9px;
}
