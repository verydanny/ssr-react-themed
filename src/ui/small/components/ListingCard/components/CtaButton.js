import React from 'react'
import PropTypes from 'prop-types'
import { themed } from 'react-themed'
import cn from 'classnames'

const CtaButton = ({
  contactType,
  link,
  tags,
  onClick,
  variation,
  theme,
  children,
  nodeType: NodeType,
  ...others
}) => (
  <NodeType
    className={cn(
      theme.CtaButton,
      theme[`CtaButton-${contactType}`],
      variation && theme[`CtaButton-${contactType}-${variation}`],
    )}
    href={link}
    onClick={onClick}
    {...tags}
    {...others}
  >
    {children}
  </NodeType>
)

CtaButton.propTypes = {
  contactType: PropTypes.string.isRequired,
  link: PropTypes.string,
  tags: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired,
  variation: PropTypes.string,
  theme: PropTypes.object,
  children: PropTypes.any,
  nodeType: PropTypes.string,
}

CtaButton.defaultProps = {
  theme: {},
  nodeType: 'button',
}

export default themed(/^CtaButton/)(CtaButton)
