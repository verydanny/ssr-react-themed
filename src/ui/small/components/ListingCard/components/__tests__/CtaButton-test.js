import React from 'react'
import { shallow, mount } from 'enzyme'
import CtaButton from '../CtaButton'

describe('ListingCard/CtaButton', () => {
  const theme = {
    CtaButton: 'CtaButton',
    'CtaButton-email-test-variation': 'CtaButton-email-test-variation',
  }

  const props = {
    contactType: 'email',
    link: 'https://www.rent.com',
    tags: { 'data-tag_item': 'testTag' },
    onClick: jest.fn(),
    icon: 'email',
    theme,
    variation: 'test-variation',
  }

  it('passes through children props', () => {
    const wrapper = shallow(
      <CtaButton {...props}>
        <div className="test-child-prop">Test Prop</div>
      </CtaButton>
    )
    const div = wrapper.find('.test-child-prop')
    expect(div.text()).toBe('Test Prop')
  })

  it('sets tagging data', () => {
    const wrapper = mount(<CtaButton {...props} />)
    const container = wrapper.find('.CtaButton')
    expect(container.prop('data-tag_item')).toBe('testTag')
  })

  it('sets the link href', () => {
    const wrapper = mount(<CtaButton {...props} />)
    expect(wrapper.find('button').prop('href')).toBe(props.link)
  })

  it('sets the onClick handler', () => {
    const wrapper = mount(<CtaButton {...props} />)
    const link = wrapper.find('button')
    link.simulate('click')
    expect(props.onClick).toBeCalled()
  })

  it('sets a variation class', () => {
    const wrapper = mount(<CtaButton {...props} />)
    const container = wrapper.find(`.CtaButton-${props.contactType}-${props.variation}`)
    expect(container.length).toBe(1)
  })

  it('sets the provided nodeType', () => {
    const anchorProps = {
      nodeType: 'a',
      ...props,
      onClick: () => {},
    }
    const wrapper = mount(<CtaButton {...anchorProps} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('a')).toHaveLength(1)
    expect(wrapper.find('button')).toHaveLength(0)
  })
})
