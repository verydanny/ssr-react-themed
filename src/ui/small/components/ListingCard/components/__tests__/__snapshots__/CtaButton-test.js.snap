// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingCard/CtaButton sets the provided nodeType 1`] = `
<Themed(CtaButton)
  contactType="email"
  icon="email"
  link="https://www.rent.com"
  nodeType="a"
  onClick={[Function]}
  tags={
    Object {
      "data-tag_item": "testTag",
    }
  }
  theme={
    Object {
      "CtaButton": "CtaButton",
      "CtaButton-email-test-variation": "CtaButton-email-test-variation",
    }
  }
  variation="test-variation"
>
  <CtaButton
    contactType="email"
    icon="email"
    link="https://www.rent.com"
    nodeType="a"
    onClick={[Function]}
    tags={
      Object {
        "data-tag_item": "testTag",
      }
    }
    theme={
      Object {
        "CtaButton": "CtaButton",
        "CtaButton-email-test-variation": "CtaButton-email-test-variation",
      }
    }
    variation="test-variation"
  >
    <a
      className="CtaButton CtaButton-email-test-variation"
      data-tag_item="testTag"
      href="https://www.rent.com"
      icon="email"
      onClick={[Function]}
    />
  </CtaButton>
</Themed(CtaButton)>
`;
